pkg/logs/processor/encoder.go

```diff
@@ -23,14 +23,17 @@ func toValidUtf8(msg []byte) string {
 	if utf8.Valid(msg) {
 		return string(msg)
 	}
-	str := make([]rune, 0, len(msg))
-	for i := range msg {
-		r, size := utf8.DecodeRune(msg[i:])
-		if r == utf8.RuneError && size == 1 {
-			str = append(str, unicode.ReplacementChar)
-		} else {
-			str = append(str, r)
-		}
+
+	var str strings.Builder
+	str.Grow(len(msg))
+
+	for len(msg) > 0 {
+		r, size := utf8.DecodeRune(msg)
+		// in case of invalid utf-8, DecodeRune returns (utf8.RuneError, 1)
+		// and since RuneError is the same as unicode.ReplacementChar
+		// no need to handle the error explicitely
```

Comment:
Looks like the linter dinged you for `explicitely`, when it should be `explicitly`.